src/test/java/com/google/opensesame/mockObjects/MockGHObjects.java

```diff
@@ -109,6 +115,52 @@ public static GHRepository createMockRepository() throws IOException {
     when(mockGHRepository.getLanguage()).thenReturn(MOCK_LANGUAGE);
     when(mockGHRepository.getName()).thenReturn(MOCK_GHREPO_NAME);
     when(mockGHRepository.listTopics()).thenReturn(MOCK_TOPICS);
+    when(mockGHRepository.getId()).thenReturn(MOCK_ID);
     return mockGHRepository;
   }
+
+  // #### MOCK GHMyself ####
+
+  /** The login of the mocked GHMyself object returned by createMockMyself() */
+  public static String MOCK_LOGIN = "MOCK_LOGIN";
+
+  /**
+   * Creates an instance of a GHMyself mocked with Mockito. The mocked object comes with the
+   * following functions predefined;
+   * <li>getLogin() -> MOCK_LOGIN
+   *
+   *     <p>NOTE: MOCK_LOGIN is a fixture in this class
+   *
+   * @return a mocked instance of a GHMyself Object
+   * @throws IOException
+   */
+  public static GHMyself createMockMyself() throws IOException {
+    // Create A GHUser Stub
+    GHMyself mockGHMyself = mock(GHMyself.class);
+    when(mockGHMyself.getLogin()).thenReturn(MOCK_LOGIN);
+    return mockGHMyself;
+  }
+
+  // #### MOCK GitHub API ####
+
+  /**
+   * Creates an instance of a GitHub mocked with Mockito. The mocked object comes with the following
+   * functions predefined;
+   * <li>getMyself() -> GHMyself returned by {@link #createMockMyself() createMockMyself}
+   * <li>getUser(String anyString) -> GHUser returned by {@link #createMockUser() createMockUser}
+   * <li>getRepositoryById(String anyString) -> GHRepository returned by {@link
+   *     #createMockRepository() createMockRepository}
+   *
+   *     <p>NOTE: all functions return fixtures defined in this class
+   *
+   * @return a mocked instance of a GitHub
+   * @throws IOException
+   */
+  public static GitHub createMockGitHub() throws IOException {
+    GitHub mockGitHub = mock(GitHub.class);
+    when(mockGitHub.getMyself()).thenReturn(createMockMyself());
+    when(mockGitHub.getUser(anyString())).thenReturn(createMockUser());
+    when(mockGitHub.getRepositoryById(anyString())).thenReturn(createMockRepository());
```

Comment:
Do you think it might be worth using the ID string provided in `getUser` and `getRepositoryById` as the ID of the mocks? Otherwise I worry about some assumptions possibly being broken where we assume the ID of the user or repo returned from one of these getters is the same as the one provided as a parameter. This can be accomplished using `thenAnswer` instead of `thenReturn` as explained [here](https://stackoverflow.com/a/36627077/9017299).

Reply:
I was thinking about that actually, I just made them more specific.